build.Dockerfile

```diff
@@ -0,0 +1,105 @@
+# Copyright 2021 Integritee AG
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#           http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This is a multi-stage docker file, where the first stage is used
+# for building and the second deploys the built application.
+
```

Comment:
The docker file that is used for building now. Uses multiple stages that can be targeted indivdually, e.g. when creating an image for testing, or deployment.
* The first and most important stage is the `builder` stage - this stage builds the worker from source.
* Stages can depend on other stages and leverage the caching mechanism of docker
* The last stage is `deployed`, which is a simple ubuntu image with all the worker binaries. This separates the deployed image cleanly from our `integritee/integritee-dev` image and reduces the image size from `3.8 GB` to `~320 MB` (uncompressed)